client/state/plugins/installed/selectors-ts.ts

```diff
@@ -351,9 +353,13 @@ export const getSiteObjectsWithoutPlugin = createSelector(
 	},
 	( state: AppState ) => [ getAllPluginsIndexedByPluginSlug( state ), getSitesItems( state ) ],
 	( state: AppState, siteIds: number[], pluginSlug: string ) => [ pluginSlug, ...siteIds ].join()
-);
+) as ( state: AppState, siteIds: number[], pluginSlug: string ) => SiteDetails[];
 
-export function getStatusForPlugin( state: AppState, siteId: number, pluginId: string ) {
+export function getStatusForPlugin(
+	state: AppState,
+	siteId: number,
+	pluginId: string
+): { [ key: string ]: string } | undefined {
```

Comment:
Although this used to return `false` and this change caused the error, I still think it is best to return `undefined` here. The function name suggests it returns the plugin status, not a boolean which indicates if the status is undefined.

A search of this code shows that this selector is only used in two components and both are TypeScript, so the explicit typing here will prevent the previous error from occurring again.